swarm/api/routes/events.py

```
@@ -353,6 +355,18 @@ async def stream_run_events(run_id: str, request: Request):
     """
+    try:
+        validate_path_component(run_id, "run_id")
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "validation_error",
+                "message": str(e),
+                "details": {"run_id": run_id},
+            },
+        )
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)